test/library/input.spec.jsx

```diff
@@ -439,27 +439,6 @@ describe('NumberFormat as input', () => {
     });
   });
 
-  it('should not call onValueChange if no formatting is applied', async () => {
-    const mockOnValueChange = vi.fn();
-
-    const { rerender } = await render(<NumericFormat value="" onValueChange={mockOnValueChange} />);
-
-    expect(mockOnValueChange).not.toHaveBeenCalled();
-
-    rerender(<NumericFormat value={NaN} onValueChange={mockOnValueChange} />);
-    expect(mockOnValueChange).not.toHaveBeenCalled();
-
-    rerender(<NumericFormat value={1234} onValueChange={mockOnValueChange} />);
-    expect(mockOnValueChange).not.toHaveBeenCalled();
-
-    rerender(<NumericFormat value={1234} onValueChange={mockOnValueChange} thousandSeparator />);
-    expect(mockOnValueChange.mock.lastCall[0]).toEqual({
-      formattedValue: '1,234',
-      value: '1234',
-      floatValue: 1234,
-    });
-  });
-
```

Comment:
This doesn't seem to be a correct test, why should `onValueChange` _not_ be called when the input goes from a `NaN` value to a meaningful value? I have tried to capture this, and further scenarios in the test I have added below, please let me know if this should be adjusted or kept instead of being removed.